### PR TITLE
Handle GitHub removing Py27 and move pytype from to Py3.8 to fix issues

### PR DESCRIPTION
**Pull request by bernhardkaindl**
Note: Because of the announcement of removal of Python 2.7 for setup-python in https://github.com/actions/setup-python/issues/672, use a wrapper for setup-python which can build missing Python versions.

### Commits:
- [run-pytype.py: Empty the list of excludes and xfails](https://github.com/xenserver/python-libs/commit/bae99960b42b5d0a952c35d1969b7af57d1892fd)

  * As pytype errors are fixed, clear excludes and xfails

- [tox.ini and CI: Move pytype to Py3.8 and remove 3.7 and 3.9 builds](https://github.com/xenserver/python-libs/pull/93/commits/2f6fdd833e4299a85223e23fe623df620e101b8c)

  - pytype has several problems to import libraries when used on Python 3.10, therefore move it Python 3.8.
  - tox.ini, GitHub CI: Remove builds for 3.7 and 3.9 builds because they showed intermittent pytest failures and we shoud not use them as new Python versions anyway
 
- [.github/workflows/main.yml: Py2.7: Use a wrapper for setup-python](https://github.com/xenserver/python-libs/pull/93/commits/a0b8a36a214dfadd428eed3e39d75622c77d75ca)

  - Because of the announcement of removal of Python 2.7 for setup-python in https://github.com/actions/setup-python/issues/672, use a wrapper for setup-python which can build missing Python versions.

### CODE REVIEW

File: run-pytype.py [hunk 1]
```diff
@@ -34,7 +32,8 @@ def run_pytype(command: List[str], branch_url: str, errorlog: TextIO, results):
                 printfrom = line.index("]") + 2
             info("PROGRESS: " + line[1:].split("]")[0] + ": " + line[printfrom:])
             continue
-        elif line.startswith("ninja: "):
+
+        if line.startswith("ninja: "):
```
Comment (psafont) [inline run-pytype.py:36]: Why was this changed?

**Review by psafont — APPROVED** [review]
hopefully python2 can be dropped soon

File: run-pytype.py (hunk 1 above)
Comment (bernhardkaindl) [reply run-pytype.py:36]: Is is changed to fix this pylint report (Using "elif" after a "continue" is not necessary):

> Unnecessary "elif" after "continue", remove the leading "el" from "elif" (no-else-continue)

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -43,9 +39,17 @@ jobs:
         with:
           fetch-depth: 0  # Needed by diff-cover to get the changed lines: origin/master..HEAD
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+        # uses: actions/setup-python@v4 due to https://github.com/actions/setup-python/issues/672:
+        # https://github.com/MatteoH2O1999/setup-python
+        # This action tries to build from source all Python versions that actions/setup-python
+        # does not support. It also allows to cache built versions so that after the first run,
+        # installation time is really low. Hope it works also with 20.04, else it needs more changes:
```
Comment (GeraldEV) [inline .github/workflows/main.yml:46]: Have we tested this with 20.04?

File: run-pytype.py [hunk 3]
```diff
@@ -107,34 +106,28 @@ def main(me: str, branch_url: str):
...
-    excludes = [
-        "xcp/cmd.py",
-        "xcp/net/ip.py",
-    ]
-    errors_in = excludes.copy()
-    errors_in.extend(never)
+    broken_files = []  # type: list[str]
+    xfail_files = []  # type: list[str]
+    errors_in = xfail_files.copy()
+    errors_in.extend(broken_files)
```
Comment (GeraldEV) [inline run-pytype.py:112]: Extending by known empty list (since it was defined as empty on 109)

File: run-pytype.py [hunk 4]
```diff
@@ -107,34 +106,28 @@ def main(me: str, branch_url: str):
...
     base = [
         "pytype",
         "-k",
         "--config",
         ".github/workflows/pytype.cfg",
     ]
     command = base.copy()
-    command.extend(["--exclude", " ".join(errors_in)])
+    if errors_in:
```
Comment (GeraldEV) [inline run-pytype.py:120]: Is there any way for errors_in to have contents at this point? It's the copy of an empty list extended by an empty list

File: run-pytype.py (hunk 3 above)
Comment (bernhardkaindl) [reply run-pytype.py]: Fixed, these variables were used for configuration, now they are read from config files!
(instead of `broken_files`, we can use the `exclude` config setting in `pytype.cfg`)

File: run-pytype.py (hunk 4 above)
Comment (bernhardkaindl) [reply run-pytype.py]: Fixed: Like said before, these are to be configuration settings:
- Improved the code to read them from configuration files.

File: .github/workflows/main.yml (hunk 2 above)
Comment (bernhardkaindl) [reply .github/workflows/main.yml]: It stopped working with GitHub's new ubuntu-20.04 rootfs.

Therefore, I moved it and the other job to ubuntu-22.04.

This means all of GitHub CI is using one distro (ubuntu-22.04) as base rootfs,
which is great! Hopefully less unpleasant surprises from GitHub on that!

**Review by GeraldEV — APPROVED** [review]
